.github/workflows/testing.yml

```diff
@@ -19,6 +19,7 @@ jobs:
           python-version: '3.10'
       - name: Install Dependencies
         run: |
+          pip install --no-cache-dir --upgrade setuptools>=77.0 
```

Comment:
Currently the build step in this workflow won't respect the bound in pyproject.toml, so a compatible version has to be installed by hand. This could be streamlined in a future PR.